test/suite/test_bug010.py

```diff
@@ -66,7 +66,8 @@ def test_checkpoint_dirty(self):
 
             # Create a checkpoint thread
             done = threading.Event()
-            ckpt = wtthread.checkpoint_thread(self.conn, done)
+            testcase = self.currentTestCase()
```

Comment:
I would rather have `testcase = WiredTigerTestCase.currentTestCase() `be done in the wtthread.py. We shouldn't have developers who utilise the wtthread functionality to know this knowledge.

Reply:
The reason I put the getter call in each test function is to make it explicit it's from the calling thread. Another option is to call the getter inside the constructor (i.e. `__init__` method) of each thread, which feels not intuitive thinking the object is actually retrieved from the caller thread's local storage. The benefit would be a more inherent design, hiding the details of testcase retrieval and setting across threads' local storage. The more I think of it the more I tend to agree with you. I can add a comment there to hopefully avoid confusion.